src/Transport/Topology/MetaModel/NamespacesDefinition.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace NServiceBus.AzureServiceBus
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class NamespacesDefinition : IEnumerable<NamespaceDefinition>
+    {
+        private static readonly string DefaultName = "default";
+        private static readonly string Prefix = "namespace-";
+
+        private readonly List<NamespaceDefinition> _inner;
```

Comment:
I think no: NamespacesDefinition is configured during endpoint and transport configuration which is a "single thread" operation (I suppose).
Once configured, NamespacesDefinition is "multi thread readable"